cmd/crates/soroban-spec-typescript/src/lib.rs

```diff
@@ -96,7 +96,12 @@ pub fn generate(spec: &[ScSpecEntry]) -> String {
             cases: vec![],
         });
     }
-    let (fns, other): (Vec<_>, Vec<_>) = collected
+    // Filter out function entries with names that start with "__"
+    let filtered_collected: Vec<_> = collected
+        .into_iter()
+        .filter(|entry| !matches!(entry, Entry::Function { name, .. } if name.starts_with("__")))
```

Comment:
```suggestion
        .filter(|entry| matches!(entry, Entry::Function { name, .. } if !name.starts_with("__")))
```
Nice! I didn't realize that if gaurds worked with the `matches` macro, which makes sense.